stl/inc/algorithm

```diff
@@ -2348,6 +2348,83 @@ _FwdIt2 replace_copy_if(_ExPo&&, _FwdIt1 _First, _FwdIt1 _Last, _FwdIt2 _Dest, _
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::fill
+    class _Fill_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <class _Ty, output_iterator<const _Ty&> _It, sentinel_for<_It> _Se>
+        constexpr _It operator()(_It _First, _Se _Last, const _Ty& _Value) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst      = _Get_unwrapped(_STD move(_First));
+            const auto _ULast = _Get_unwrapped(_STD move(_Last));
+            if constexpr (_Fill_memset_is_safe<decltype(_UFirst), _Ty>) {
+#ifdef __cpp_lib_is_constant_evaluated
+                if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+                {
+                    _CSTD memset(_UFirst, static_cast<unsigned char>(_Value), static_cast<size_t>(_ULast - _UFirst));
+                    _UFirst += static_cast<size_t>(_ULast - _UFirst);
+                    _Seek_wrapped(_First, _UFirst);
+                    return _First;
+                }
+            }
+
+            while (_UFirst != _ULast) {
+                *_UFirst = _Value;
+                ++_UFirst;
+            }

```

Comment:
Style: This is plain iteration, so would it be better to write this as a for-loop? Usually `while` loops should be used for unusual iteration (trying to cram unusual logic into a for-loop is counterproductive), but here we're just incrementing all the way to the end. A for-loop without an initializer is reasonable, we have plenty of those. Then this would look more like `fill_n` below.